boards/MicroBit/src/microbit-i2c.adb

```diff
@@ -52,6 +52,14 @@ package body MicroBit.I2C is
    ----------------
 
    procedure Initialize (S : Speed := S400kbps) is
+      Config   : GPIO_Configuration := (Mode => Mode_In,
```

Comment:
`Config` should be `constant` as it is not modified.

Also you need to add `use nRF.GPIO` for the type to be visible.